catppuccin.tmux

```diff
@@ -1,23 +1,159 @@
+
+  # windows
+  setw window-status-activity-style "fg=${thm_fg},bg=${thm_bg},none"
+  setw window-status-separator ""
+  setw window-status-style "fg=${thm_fg},bg=${thm_bg},none"
+
+  # --------=== Statusline
+
+  # NOTE: Checking for the value of @catppuccin_window_tabs_enabled
+  wt_enabled="$(get-tmux-option "@catppuccin_window_tabs_enabled" "off")"
+  readonly wt_enabled
+
+  # NOTE: Checking for the value of @catppuccin_window_tabs_enabled
+  wi_enabled="$(get-tmux-option "@catppuccin_window_icons_enabled" "off")"
+  readonly wi_enabled
+
+  # README: Any variables marked `readonly` are the defaults so that it is
+  # easier to understand what is a default value. Some variables are marked
+  # `local` and that's because they will be used in place for the `setw` and
+  # `set` commands to make it easier to parse what is happening.
+
+  # Default Nerd Font zoom/mark icons
+  readonly icon_window_current="  "
+  readonly icon_window_last="  "
+  readonly icon_window_activity="  "
+  readonly icon_window_bell=" "
+  readonly icon_window_silent=" "
+  readonly icon_window_zoom="  "
+  readonly icon_window_mark=" "
+
+  # README: The following variables are set to the defaults above but can be
+  # overwritten by the user.
+  custom_icon_window_last="$(get-tmux-option "@catppuccin_icon_window_last" "${icon_window_last}")"
+  readonly custom_icon_window_last
+
+  custom_icon_window_current="$(get-tmux-option "@catppuccin_icon_window_current" "${icon_window_current}")"
+  readonly custom_icon_window_current
+
+  custom_icon_window_zoom="$(get-tmux-option "@catppuccin_icon_window_zoom" "${icon_window_zoom}")"
+  readonly custom_icon_window_zoom
+
+  custom_icon_window_mark="$(get-tmux-option "@catppuccin_icon_window_mark" "${icon_window_mark}")"
+  readonly custom_icon_window_mark
+
+  custom_icon_window_silent="$(get-tmux-option "@catppuccin_icon_window_silent" "${icon_window_silent}")"
+  readonly custom_icon_window_silent
+
+  custom_icon_window_activity="$(get-tmux-option "@catppuccin_icon_window_activity" "${icon_window_activity}")"
+  readonly custom_icon_window_activity
+
+  custom_icon_window_bell="$(get-tmux-option "@catppuccin_icon_window_bell" "${icon_window_bell}")"
+  readonly custom_icon_window_bell
+
+  # Set default the window status variable
+  local show_window_status="#F "
+
+  # NOTE: With the @catppuccin_window_icons_enabled set to on, we're going to
+  # update the show_window_status with a custom command to replace all the
+  # things that it would normally use according to the Tmux source.
+  if [[ "${wi_enabled}" == "on" ]]
+  then
+    show_window_status="#(printf '%%s\n' '#F' | sed \"s/*/${custom_icon_window_current}/\" | sed \"s/-/${custom_icon_window_last}/\" | sed \"s/#/${custom_icon_window_activity}/\" | sed \"s/#//g\"| sed \"s/~/${custom_icon_window_silent}/\" | sed \"s/!/${custom_icon_window_bell}/\" | sed \"s/M/${custom_icon_window_mark}/\" | sed \"s/Z/${custom_icon_window_zoom}/\")"
```

Comment:
I am using `sed` here instead of `tr` because I need the spaces used in the definition of the icons from either the defaults in this file or the icons that the user supplies in order to give the granular control we need to display them properly. I tested and saw that `sed` and `tr` were very close in terms of speed.

Reply:
You can do all the replacements at once:
`show_window_status="#(printf '%%s\n' '#F' | sed \"s/*/${custom_icon_window_current}/; s/-/${custom_icon_window_last}/; s/#/${custom_icon_window_activity}/; s/~/${custom_icon_window_silent}/; s/!/${custom_icon_window_bell}/; s/!/${custom_icon_window_bell}/; s/M/${custom_icon_window_mark}/; s/Z/${custom_icon_window_zoom}/\")"`